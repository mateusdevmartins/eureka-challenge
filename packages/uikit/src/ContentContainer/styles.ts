import styled from 'styled-components';

export const OutterView = styled.div`
  border-radius: 4px;
  display: flex;
  background-color: #ffffff;
  margin: 0px 22px 0px 22px;
  height: 386px;
  width: 33%;
  @media (max-width: 1100px) {
    width: 90%;
    margin-bottom: 67px;
  }
`;

export const InnerView = styled.div`
  display: flex;
  overflow-y: auto;
  scrollbar-width: auto;
  flex: 1;
  margin: 10px 10px 10px 10px;
  padding: 0px 16px 0px 16px;
  background-color: #ffffff;
  flex-direction: column;
`;
