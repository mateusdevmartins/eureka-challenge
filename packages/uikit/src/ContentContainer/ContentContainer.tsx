import { ReactNode } from 'react';
import { InnerView, OutterView } from './styles';

export interface ContentContainerProps {
  children: ReactNode;
}

export function ContentContainer({ children }: ContentContainerProps) {
  return (
    <OutterView data-testid="outter-view">
      <InnerView data-testid="inner-view">{children}</InnerView>
    </OutterView>
  );
}
