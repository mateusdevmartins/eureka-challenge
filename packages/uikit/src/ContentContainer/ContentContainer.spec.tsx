import renderer from 'react-test-renderer';
import { ContentContainer } from './ContentContainer';

import { render, fireEvent, getByTestId } from '@testing-library/react';

describe('ContentContainer', () => {
  it(' should match snapshot', () => {
    const component = renderer.create(
      <ContentContainer>
        <div></div>
      </ContentContainer>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('behavior', () => {
    const components = render(
      <ContentContainer>
        <div></div>
      </ContentContainer>
    );
    const { getByTestId } = components;

    expect(getByTestId('outter-view')).toBeTruthy();
    expect(getByTestId('outter-view').children.length).toBe(1);
    expect(getByTestId('inner-view')).toBeTruthy();
    expect(getByTestId('inner-view').children.length).toBe(1);
  });
});
