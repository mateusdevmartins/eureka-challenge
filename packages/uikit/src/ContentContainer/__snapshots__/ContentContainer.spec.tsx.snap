// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer  should match snapshot 1`] = `
<div
  className="styles__OutterView-m7ssp0-0 cFOpin"
  data-testid="outter-view"
>
  <div
    className="styles__InnerView-m7ssp0-1 zdZa-d"
    data-testid="inner-view"
  >
    <div />
  </div>
</div>
`;
